### PR TITLE
Migrate HwInfo data to systems

### PR DESCRIPTION
**Pull request by ngetahun**
## Description

This PR adds data migration from `hw_infos` table to `systems.system_information` json column.
To test: 
```bash
#Grab system dump file from the trello card
(mariadb)> mysql -u rmt -D rmt -p < systems_dump
(rmt)> bin/rails db:migrate
# To validate, check if systems_information column is not empty
(rmt)> bin/rails c
(rmt)> System.where(system_information: nil).count # should be 0 
```

## Change Type

*Please select the correct option.*

- [ ] **Bug Fix** (a non-breaking change which fixes an issue)
- [x] **New Feature** (a non-breaking change which adds new functionality)
- [ ] **Documentation Update** (a change which only updates documentation)

## Checklist

*Please check off each item if the requirement is met.*

- [x] I have verified that my code follows RMT's coding standards with `rubocop`.
- [x] I have reviewed my own code and believe that it's ready for an external review.
- [x] I have provided comments for any hard-to-understand code.
- [ ] I have documented the `MANUAL.md` file with any changes to the user experience.
- [x] RMT's test coverage remains at 100%.
- [ ] If my changes are non-trivial, I have added a changelog entry to notify users at `package/obs/rmt-server.changes`.

## Other Notes

Please use this space to provide notes or thoughts to the team, such as tips on how to review/demo your changes.

### CODE REVIEW

**Comment by ngetahun** [comment]
[[CMD][RMT] Migrate existing data from old HwInfo to new backend attributes](https://trello.com/c/VLdz5Lbe/2958-cmdrmt-migrate-existing-data-from-old-hwinfo-to-new-backend-attributes)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/327810/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged de57c8ab31631c9fcf91fb97f89eed5eac29bd10 into eea5e93c4380bed7bd0d06e7ef8439dd8d022118)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/327935/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged de57c8ab31631c9fcf91fb97f89eed5eac29bd10 into eea5e93c4380bed7bd0d06e7ef8439dd8d022118)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/328288/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged de57c8ab31631c9fcf91fb97f89eed5eac29bd10 into eea5e93c4380bed7bd0d06e7ef8439dd8d022118)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/328427/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged de57c8ab31631c9fcf91fb97f89eed5eac29bd10 into eea5e93c4380bed7bd0d06e7ef8439dd8d022118)

File: db/migrate/20230814105634_move_hw_info_to_systems_table.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class MoveHwInfoToSystemsTable < ActiveRecord::Migration[6.1]
+  def up
+    safety_assured do
+      execute "update systems as s inner join hw_infos hw on s.id=hw.system_id \
+                set system_information = json_object(\
```
Comment (lagartoflojo) [inline db/migrate/20230814105634_move_hw_info_to_systems_table.rb:5]: In SCC, the JSONB field is called "properties", but in RMT it's called "system_information"? Why the discrepancy?

Comment (ngetahun) [reply db/migrate/20230814105634_move_hw_info_to_systems_table.rb:5]: I don't know why that was decided, perhaps @felixsch might be able to answer

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/328455/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 46827b6c901721426b3a0ef3b1c95e7cb20f8d68 into eea5e93c4380bed7bd0d06e7ef8439dd8d022118)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/328459/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 155bec4a288195426fd29e0c64347dddedd1d2c0 into eea5e93c4380bed7bd0d06e7ef8439dd8d022118)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/328481/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 155bec4a288195426fd29e0c64347dddedd1d2c0 into b0d07472dfd20c1276d6ff75728b34dd0be4b73a)

File: db/migrate/20230814105634_move_hw_info_to_systems_table.rb (hunk 1 above)
Comment (felixsch) [reply db/migrate/20230814105634_move_hw_info_to_systems_table.rb:5]: This is due to the fact that this column **only** ever stores system related information. It is not a generic storage column as in SCC therefore a qualified name suits better then a generic one.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/328921/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 6cafc8c0548e887d19dedc9758031518f9575ba6 into b0d07472dfd20c1276d6ff75728b34dd0be4b73a)

File: db/migrate/20230814105634_move_hw_info_to_systems_table.rb (hunk 1 above)
Comment (lagartoflojo) [reply db/migrate/20230814105634_move_hw_info_to_systems_table.rb:4]: I see, thanks!

**Comment by mssola** [comment]
@ngetahun I took the liberty to fix the strong_migration issues myself. Also, sorry for the mess with creating a commit and then reverting it back, I understood the movement from strong_migration wrong.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/328922/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 2c9b7a067f143d7a8c8311d194ae3d94e75c98ff into b0d07472dfd20c1276d6ff75728b34dd0be4b73a)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/328923/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 25e9fdb819fe554435f2f6877e35599e8ab363da into b0d07472dfd20c1276d6ff75728b34dd0be4b73a)

**Review by mssola — APPROVED** [review]

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/329099/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 25e9fdb819fe554435f2f6877e35599e8ab363da into 12d5be41a920e2555e8518b85f139eed7fc557d6)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/329101/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 90441175739dfda11ebd4394b20e8a20c0676dca into 6accd59d14893ac9b35841eb6f41d565675916b2)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/329249/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 90441175739dfda11ebd4394b20e8a20c0676dca into 6accd59d14893ac9b35841eb6f41d565675916b2)

**Pull request merged**